### PR TITLE
tfa-layerscape: fix compilation with GCC11

### PR DESCRIPTION
**Pull request by neheb**
Assignment to rt_svc_descs with = doesn't work as intended as GCC11
seems to assume a [0] array for some reason. Work around with memcpy to
clarity that MAX_RT_SVCS is the correct array size.

Also remove const from pointer type as memcpy doesn't work with const.
It's not correct anyway.

Refresh other patch.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

ping @aparcar

### CODE REVIEW

**Comment by neheb** [comment]
Huh. restool actually has an issue specific to GCC11. Easily fixed.

**Comment by adschm** [comment]
@yangbolu1991 FYI

File: package/network/utils/layerscape/restool/patches/010-gcc11.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/restool.h
++++ b/restool.h
+@@ -60,7 +60,7 @@
+ /**
+  * Max length of a device file
+  */
+-#define DEV_FILE_SIZE		15
++#define DEV_FILE_SIZE		16
```
Comment (hauke) [inline package/network/utils/layerscape/restool/patches/010-gcc11.patch:8]: This is also part of this upstream patch:
https://source.codeaurora.org/external/qoriq/qoriq-components/restool/commit/?h=la1224rdb-early-access&id=91b82e3db29b694e4445c23e5930b379064e884b

This also fixes some other problems with gcc 10.2

Comment (neheb) [reply package/network/utils/layerscape/restool/patches/010-gcc11.patch:8]: Doesn't apply cleanly. Maybe it's better to update restool...

**Comment by aparcar** [comment]
Please rebase.

**Comment by neheb** [comment]
Rebased and updated restool instead. Upstream already fixed all the GCC 10 and 11 issues.

**Comment by aparcar** [comment]
> all

Can we remove this too? package/boot/tfa-layerscape/patches/002-gcc11.patch

**Comment by neheb** [comment]
No that's a different package.

**Comment by neheb** [comment]
This PR was changed to just update the packages to LSDK-20.12. Upstream fixed this.

**Comment by aparcar** [comment]
Merged, thanks!

**Pull request closed without merging**